frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Table, TextFilter } from '@console/internal/components/factory';
+import { SortByDirection } from '@patternfly/react-table';
+import { CheckBoxes } from '@console/internal/components/row-filter';
+import { getQueryArgument } from '@console/internal/components/utils';
+import {
+  CustomResourceListFilterType,
+  CustomResourceListProps,
+} from './custom-resource-list-types';
+
+const CustomResourceList: React.FC<CustomResourceListProps> = ({
+  dependentResource,
+  items,
+  queryArg,
+  rowFilters,
+  getFilteredItems,
+  resourceHeader,
+  resourceRow,
+}) => {
+  const [listItems, setListItems] = React.useState([]);
+  const [filteredListItems, setFilteredListItems] = React.useState([]);
+  const [fetched, setFetched] = React.useState(false);
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const queryArgument = getQueryArgument(queryArg);
+    const activeFilters = queryArgument?.split(',');
+
+    const fetchListItems = async () => {
+      let newListItems: any;
+      try {
+        newListItems = await items;
+      } catch {
+        if (ignore) return;
+
+        setListItems([]);
+        setFetched(true);
+      }
+
+      if (ignore) return;
+
+      setListItems(newListItems || []);
+      setFetched(true);
+
+      if (activeFilters) {
+        const filteredItems = () => {
+          return getFilteredItems(newListItems, CustomResourceListFilterType.Row, activeFilters);
+        };
+        setFilteredListItems(filteredItems);
+      } else {
+        setFilteredListItems(newListItems);
+      }
+    };
+
+    fetchListItems();
+
+    return () => {
+      ignore = true;
+    };
+  }, [dependentResource, items, queryArg, rowFilters, getFilteredItems]);
+
+  const applyRowFilter = React.useCallback(
+    (filter) => {
+      const filteredItems = getFilteredItems(listItems, CustomResourceListFilterType.Row, filter);
```

Comment:
This makes it very specific to the component. I would prefer if the parent just sent `rowFilterReducer`.